lib/commons/color/element-is-distinct.js

```diff
@@ -24,64 +11,105 @@ function _getFonts(style) {
  * @param  {HTMLElement} ancestorNode The ancestor node element to check
  * @return {Boolean}
  */
-function elementIsDistinct(node, ancestorNode) {
-  var nodeStyle = window.getComputedStyle(node);
+export default function elementIsDistinct(node, ancestorNode) {
+  const nodeStyle = window.getComputedStyle(node);
 
-  // Check if the link has a background
-  if (nodeStyle.getPropertyValue('background-image') !== 'none') {
-    return true;
-  }
+  const ancestorStyle = window.getComputedStyle(ancestorNode);
+  let curNode = node;
+  while (curNode !== ancestorNode) {
+    const curNodeStyle = window.getComputedStyle(curNode);
 
-  // Check if the link has a border or outline
-  var hasBorder = ['border-bottom', 'border-top', 'outline'].reduce(
-    (result, edge) => {
-      var borderClr = new Color();
-      borderClr.parseString(nodeStyle.getPropertyValue(edge + '-color'));
-
-      // Check if a border/outline was specified
-      return (
-        result ||
-        // or if the current border edge / outline
-        (nodeStyle.getPropertyValue(edge + '-style') !== 'none' &&
-          parseFloat(nodeStyle.getPropertyValue(edge + '-width')) > 0 &&
-          borderClr.alpha !== 0)
-      );
-    },
-    false
-  );
-
-  if (hasBorder) {
-    return true;
+    // Check if the link has a background
+    if (curNodeStyle.getPropertyValue('background-image') !== 'none') {
+      return true;
+    }
+
+    if (
+      hasSameText(node, curNode) &&
+      (hasVisibleBorder(curNodeStyle) ||
+        hasVisibleTextDecortation(curNodeStyle, ancestorStyle))
+    ) {
+      return true;
+    }
+
+    curNode = getComposedParent(curNode);
   }
 
-  var parentStyle = window.getComputedStyle(ancestorNode);
-  // Compare fonts
-  if (_getFonts(nodeStyle)[0] !== _getFonts(parentStyle)[0]) {
+  // font styles are inherited so we only need to check
+  // the target node
+  if (getFonts(nodeStyle)[0] !== getFonts(ancestorStyle)[0]) {
     return true;
   }
 
-  var hasStyle = [
-    'text-decoration-line',
-    'text-decoration-style',
-    'font-weight',
-    'font-style',
-    'font-size'
-  ].reduce((result, cssProp) => {
+  let hasStyle = ['font-weight', 'font-style', 'font-size'].some(cssProp => {
```

Comment:
Maybe this stuff that inherits should go above the while loop. Also, a comment here would be nice to indicate this isn't in the while loop because it inherits. After not having thought about this for three weeks, it took me a bit to realise why you did it this way.